File: musics/make-index.js
// make-index.js
const fs = require('fs');
const path = require('path');

const ROOT = path.join(__dirname, 'musics');     // ./musics
const OUT  = path.join(ROOT, 'index.json');

function listDir(dir) {
  return fs.readdirSync(dir, { withFileTypes: true });
}

const folders = [];
for (const entry of listDir(ROOT)) {
  if (!entry.isDirectory()) continue;
  const folderName = entry.name;
  const abs = path.join(ROOT, folderName);

  const tracks = listDir(abs)
    .filter(d => d.isFile() && /\.(mp3|m4a|aac|ogg|wav)$/i.test(d.name))
    .map((f) => ({
      title: path.parse(f.name).name,
      artist: "Radio Bang",
      file: `${folderName}/${f.name}`
      // optionally: cover: "image/logo.png"
    }));

  if (tracks.length) folders.push({ name: folderName, tracks });
}

const json = { folders };
fs.writeFileSync(OUT, JSON.stringify(json, null, 2), 'utf8');
console.log(`✅ Wrote ${OUT} with ${folders.length} folder(s).`);

File: musics/make_index.py
# make_index.py
import os, json

ROOT = os.path.join(os.getcwd(), "musics")   # ./musics
OUT  = os.path.join(ROOT, "index.json")

folders = []
for name in sorted(os.listdir(ROOT)):
    p = os.path.join(ROOT, name)
    if not os.path.isdir(p): continue
    tracks = []
    for f in sorted(os.listdir(p)):
        if f.lower().endswith((".mp3",".m4a",".aac",".ogg",".wav")):
            title = os.path.splitext(f)[0]
            tracks.append({
                "title": title,
                "artist": "Radio Bang",
                "file": f"{name}/{f}"
                # "cover": "image/logo.png"
            })
    if tracks:
        folders.append({"name": name, "tracks": tracks})

with open(OUT, "w", encoding="utf-8") as fh:
    json.dump({"folders": folders}, fh, ensure_ascii=False, indent=2)

print(f"✅ Wrote {OUT} with {len(folders)} folder(s).")
